jstest/__main__.py

```diff
@@ -166,22 +168,28 @@ def main():
     if os.environ.get('VERBOSE', '') and env['info']['quiet']:
         print('\n\033[1;33mWarning: --quiet option disables VERBOSE output!\033[0m\n')
 
-    # Initialize the testing environment by building all the
-    # required modules to be ready to run tests.
-    builder = jstest.builder.create(env)
-    builder.create_profile_builds()
-    builder.create_test_build()
-
-    # Run all the tests.
-    # FIXME this will have to remain in an if block until
-    # dummy devices are created for the Travis jobs.
-    if not env['info']['no_test']:
-        testrunner = jstest.testrunner.TestRunner(env)
-        testrunner.run()
-        testrunner.save()
-
-        if env['info']['emulate']:
-            jstest.emulate.pseudo_terminal.close_pseudo_terminal(env)
+    try:
+        # Initialize the testing environment by building all the
+        # required modules to be ready to run tests.
+        builder = jstest.builder.create(env)
+        builder.create_profile_builds()
+        builder.create_test_build()
+
+        # Run all the tests.
+        # FIXME this will have to remain in an if block until
+        # dummy devices are created for the Travis jobs.
+        if not env['info']['no_test']:
+            testrunner = jstest.testrunner.TestRunner(env)
+            testrunner.run()
+            testrunner.save()
+
+            if env['info']['emulate']:
+                jstest.emulate.pseudo_terminal.close_pseudo_terminal(env)
+    except (Exception, KeyboardInterrupt) as e:
+        console.fail('[Failed] %s' % (str(e)))
```

Comment:
Please use `console.log` instead of `console.error`.

Reply:
Note: there are some places where `console.fail` is used. Since `console.fail` exits the program you should modify that method to throw an exception instead.